lib/connectors_async/job_store.rb

```diff
@@ -0,0 +1,37 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+
+require 'concurrent'
+require 'securerandom'
+
+require 'connectors_async/job'
+
+module ConnectorsAsync
+  class JobStore
+    class JobNotFoundError < StandardError; end
+
+    def initialize
+      # multiple threads can write to the store
+      @store = Concurrent::Hash.new
+    end
+
+    def create_job
+      job_id = SecureRandom.uuid
+
+      job = ConnectorsAsync::Job.new(job_id)
+
+      @store[job_id] = job
```

Comment:
Do we ever remove old jobs from `@store`?

Reply:
Yes, I believe that we only keep job history for ~30 days? I'm not quickly finding where that job history deletion logic lives though.

Reply:
Agreed. `job_store` should be able to clean finished/stale jobs. e.g. 
```
@store.delete(job.id) if job.is_finished? || job.stale?
```

In class `Job`
```
def stale?
  Time.now - last_updated > 5.mins
end
```

Reply:
Not yet, I decided not to put the job cleanup in this PR and handle it in a different PR (https://github.com/elastic/enterprise-search-team/issues/1980) - same goes for Token Refresh (https://github.com/elastic/enterprise-search-team/issues/1979)